doc/user-guide/embed-sql-py.md

```diff
@@ -0,0 +1,73 @@
+---
+jupytext:
+  notebook_metadata_filter: myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+myst:
+  html_meta:
+    description lang=en: Embed SQL queries in .py file
+    keywords: jupyter, sql, jupysql
+    property=og:locale: en_US
+---
+
+# Cell magics in .py scripts
+
+Several users may want to run SQL queries in a Python script instead of Jupyter Notebook. In this tutorial, we'll demonstrate how to embed JupySQL magics in a Python file using VSCode.
+
+## Python Interactive Window
+
+VSCode allows users to work with Jupyter-like code cells and run code in the Python Interactive Window. To work with these code cells, first, select the Python environment in which JupySQL is installed. To select an environment, use the **Python: Select Interpreter** command from the Command Palette.
+
+Once done, you can define Jupyter-like code cells within Python code using a `# %%` comment. For more details, refer [VSCode Jupyter support](https://code.visualstudio.com/docs/python/jupyter-support-py).
+
+Here's a code snippet that allows users to download a sample dataset and perform SQL queries on the data using JupySQL's `%%sql` cell magic.
+
+## Sample code
+
+```python
+# %%
+%pip install jupysql duckdb duckdb-engine --quiet
+%load_ext sql
+%sql duckdb://
+
+# %%
+from urllib.request import urlretrieve
+
+_ = urlretrieve(
+    "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv",
+    "penguins.csv",
+)
+
+# %%
+%%sql
+SELECT *
+FROM penguins.csv
+LIMIT 3
+```
+
+Now let's look at the steps for running this code in VSCode.
+
+First, create a new file and select the file type as `Python File` as shown below:
+
+![file type](../static/vscode-file-type.png)
+
+Now, add a code cell and try to run the cell. It would prompt the user to install the `ipykernel`.
+
+![file ipykernel](../static/vscode-ipykernel.png)
+
+Ensure to select the correct Python environment for the code cell to run properly:
+
+![env](../static/vscode-env.png)
+
+Now, run the file in the interactive mode as shown below. You may also run each cell individually by clicking the `Run Cell` option.
+
+![run_interactive](../static/vscode-run-interactive.png)
+
```

Comment:
One question I foresee coming up is: does this workflow still hold if I were to containerize my .py script and execute outside of VSCode? What about other IDEs?

Reply:
I think we can keep it as is - as long as it works on VSCode that's fine.

But we can also mention that this will work on Spyder and PyCharm since they also support the percent format.

Another thing we can mention is jupytext and [ploomber-engine](https://engine.ploomber.io/en/latest/quick-start.html). Since users might be interested in running this scripts programmatically

```
jupytext sql-analysis.py --to ipynb
ploomber-engine sql-analysis.ipynb output.ipynb
```

Reply:
Have added Spyder, but I couldnt find run interactive cell in PyCharm. I'll check again
Adding `ploomber-engine`

Reply:
related: https://github.com/mwouts/jupytext/issues/153

Reply:
Added the above points